daemon/logger/copier.go

```diff
@@ -112,7 +136,20 @@ func (c *Copier) copySrc(name string, src io.Reader) {
 					msg.Source = name
 					msg.Timestamp = time.Now().UTC()
 					msg.Line = append(msg.Line, buf[p:n]...)
-					msg.Partial = true
+
+					// Generate unique partialID for first partial. Use it across partials.
+					// Initialize index for first partial. Increment it across partials.
+					if firstPartial {
+						partialid = stringid.GenerateRandomID()
+						ordinal = 0
+						firstPartial = false
+					}
+					ordinal++
+					setLogMetaData(msg, partialid, ordinal, false)
```

Comment:
Looking at the code, I'm a bit in doubt if we should keep the `setLogMetaData()` function, or if it's clearer to just do it inline.

Was even looking if it's possible somehow to do `msg := NewMessage()` if we create a new log-entry, and keep the same `msg`, until we reached the end. This would allow

- Generating a unique ID inside `NewMessage()` - both partial and non-partial log-entries would get an ID
- Inside the loop do `msg.MetaData.ordinal++`
- no need keep track  / "reset" these local variables (`msg = NewMessage()` would reset them)

Haven't fully gone through that idea though to see if that's easy to refactor

Reply:
The design is such that IDs are handed out only to partials. Full log entries will have empty IDs.

Reply:
Right, so wondering if it would be easier to process if an ID was always present (basically, a "full" entry is just a partial consisting of one part)

Reply:
Not necessary. An empty ID string is an indicator for a full log entry. Another way to do this would be to have a pointer to `PartialLogMetaData` and check for `nil` pointer to indicate a full log entry.

Reply:
Let me try explaining my train of thought: if each message has a unique ID (irregardless of being partial or "full"), an aggregator could find all parts of a message by "grouping" by ID, and stitch them together based on `ordinal`.

The same logic could be used both for _partial_ and _full_ messages, so processing could be easier to handle if there's less differences between those